### DISCUSSION TITLE
Automatic reply with "most relevant posts"

### DISCUSSION THREAD

**Discussion by hugary1995** [idaholab/moose#24823] [Q&A General]
So just now I came up with this idea of letting a bot reply to every post with "the most relevant posts" before @GiudGiud gets to every one of them. I did a little bit of search online, and I found that this is very much doable. The steps are

1. Collect all the discussion posts, comments, and replies. Github GraphQL can do this very easily. I'm attaching an example query json in the end.
2. Format useful information into a dataset. Say username, title, body text, etc.
3. Use a popular model to extract features and compute post similarities. scipy can do this fairly easily, for example.
4. Set up a github bot to automatically reply every new post with the most similar posts.

Apparently, we need to periodically re-index database by running steps 1-3, but that should be an easy job for CIVET. Any ideas/suggestions, or anyone wants to collaborate on this fun little weekend project?

```yaml
query { 
  rateLimit {
    limit
    cost
    remaining
    resetAt
  }
  repository(owner: "idaholab", name: "moose") {
    discussions(first: 10) {
      pageInfo {
        hasNextPage
        endCursor
      }
      edges {
        node {
          title
          author {
            login
          }
          bodyText
          url
          updatedAt
          publishedAt
          category {
            name
          }
          comments(first: 100) {
            edges {
              node {
                author {
                  login
                }
                bodyText
                url
                updatedAt
                publishedAt
                isAnswer
                replies(first: 100) {
                  edges {
                    node {
                      author {
                        login
                      }
                      bodyText
                      url
                      updatedAt
                      publishedAt
                      isAnswer
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```

**Comment by dschwen** [comment]
Are you using word2vec?

**Reply by hugary1995** [reply]
I'm thinking of doc2vec from gensim. But ya, basically the same thing.

**Comment by MengnanLi91** [comment]
I found this github bot (https://github.com/dhrubasaha08/Discuss-Bot). We have open-source LLM on INL HPC and a rag pipeline ready. I would love to try this out and see how it works
